File: api/src/EventHub.Domain/ReadModels/DashboardOverviewDto.cs
namespace EventHub.Domain.ReadModels;

/// <summary>KPIs du tableau de bord back office (GET /api/admin/dashboard/overview).</summary>
public sealed record DashboardOverviewDto
{
    public int TotalUsers { get; init; }
    public int TotalActivities { get; init; }
    public int PublishedActivities { get; init; }
    public int UpcomingActivities { get; init; }
    public int TotalRegistrations { get; init; }
    public int WaitlistedRegistrations { get; init; }
    public long TotalHeartsAwarded { get; init; }
    public int TotalPosts { get; init; }
}

File: backoffice/src/app/features/dashboard/models/dashboardOverviewDto.ts
/** KPIs du tableau de bord (GET /api/admin/dashboard/overview). */
export interface DashboardOverviewDto {
  totalUsers: number;
  totalActivities: number;
  publishedActivities: number;
  upcomingActivities: number;
  totalRegistrations: number;
  waitlistedRegistrations: number;
  totalHeartsAwarded: number;
  totalPosts: number;
}
